### DISCUSSION TITLE
Server closes unexpectedly

### DISCUSSION THREAD

**Discussion by AliE89** [Kitware/trame#535] [General]
Hello Sebastien,

I know this is very tricky, but recently I am experimenting the situation where there are no errors on the python side, but the server opens and closes itself instantly and randomly. I am not using any timeout at the moment.

Is there any hidden log or parameter to pass in order to investigate the issue?

Thank you!

**Comment by jourdain** [comment]
Is the network getting proxied (nginx, else?)?
Do you have computation that keep the server busy more that 30s in a row?

**Comment by AliE89** [comment]
Yes,

I have some intensive computation done excedding 30s (data loading +
renders setups), but sometimes the server goes through, sometimes it
doesn't, I am wondering if there something to force it to stay up...


Tha ks

Il Gio 6 Giu 2024, 16:36 Sebastien Jourdain (Kitware) <
***@***.***> ha scritto:

**Comment by jourdain** [comment]
__Environments variables__

- __TRAME_WS_HEART_BEAT__ : Time in second before assuming the server is non-responsive. (default: 30s)

**Comment by AliE89** [comment]
Ok, I guess I just do:

export *TRAME_WS_HEART_BEAT=720*

Cheers

**Comment by AliE89** [comment]
EDIT: Also is this option useful to monitor what's going on?

   - export TRAME_LOG_NETWORK='/home/logs/my_log' *

Cheers

**Reply by jourdain** [reply]
TRAME_LOG_NETWORK is really to debug otherwise it will affect the performances

**Comment by banesullivan-kobold** [comment]
I've been running into this recently myself and a few internal users of a trame application that is hosted on JupyterHub as a named server registered with jupyter-server-proxy (ref [these docs ](https://jupyter-server-proxy.readthedocs.io/en/latest/server-process.html), basically its a nice mechanism for exposing apps under their own set of named URL paths on JupyterLab). We're seeing an issue where running `python app.py --server` eventually times out if left in a background browser tab for ~1+ hour. 

@jourdain, is using `TRAME_WS_HEART_BEAT=700` a good solution to this? Ideally we want the Trame server to remain live no matter what and let JupyterHub itself manage spinning down inactive servers. Is there a better way to make sure this process stays live? Also, we often want the server to remain live if the user closes their browser tab so that they can easily navigate back to it after a couple of hours

**Reply by jourdain** [reply]
In your usecase @banesullivan-kobold , you just need to start the trame app like so `python app.py --server --timeout 0`

**Reply by banesullivan-kobold** [reply]
Aha! Well now it's obvious! Thanks, @jourdain!
